packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -27,6 +27,7 @@ export function DialogTrigger(props: SpectrumDialogTriggerProps) {
     mobileType = type === 'popover' ? 'modal' : type,
     hideArrow,
     targetRef,
+    size,
```

Comment:
Instead of adding a size prop to this, can we take advantage of type? We could have types: modal, popover, tray, fullscreen, fullscreenTakeover. 
Make sense?

Reply:
Can you also add those 2 new types to `mobileType`. We want to allow people to override some default behaviors regarding mobile views.

Reply:
Ehh, I considered this before and it feels weird to me since it dilutes the intuitive meaning of the `type` prop. To me `type` indicates the kinda of shape/look of the Dialog where the `size` prop takes the literal meaning. This would mean a user could provide a `size` to dialog (S/M/L) and specify `type=fullscreen` as well which doesn't really make sense as a pairing

Reply:
If we are trying to distinguish that `fullscreen`/`fullscreenTakeover` is for modal only, we could go one step further and change `size` to be `modalSize` since it only really applies to  the `modal` type as is.

Reply:
yeah mostly the problem here is that DialogTrigger handles too much anyway. The situation you describe will exist anyway in other forms because we are mixing popover and modal behaviors. We will need in on mobileType so it seemed the best to just add it to type as well. Otherwise based on your suggestion we would need modalSize and mobileModalSize?

Reply:
yup we would need to add mobileModalSize as well in case the user wants separate size behavior on mobile vs desktop. 

I'll go with your suggestion for now and add the 2 fullscreen types

Reply:
ok cool